### PR TITLE
Switch from `rlang::abort()` in `parameters.R`, `grids.R` and `misc.R`

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
Ref: https://github.com/tidymodels/dials/issues/311

This PR changes `rlang::abort()` to `cli::cli_abort()` in `parameters.R`, `grids.R` and `misc.R`.

### CODE REVIEW

File: R/grids.R [hunk 1]
```diff
@@ -252,7 +253,9 @@ make_random_grid <- function(...,
 
 new_param_grid <- function(x = new_data_frame()) {
   if (!is.data.frame(x)) {
-    rlang::abort("`x` must be a data frame to construct a new grid from.")
+    cli::cli_abort(
+      "{.arg x} must be a data frame to construct a new grid from."
+    )
```
Comment (hfrick) [inline R/grids.R:258]: Can you thread the call through and give this message a test?

File: R/parameters.R [hunk 2]
```diff
@@ -12,11 +12,8 @@ parameters <- function(x, ...) {
 #' @export
 #' @rdname parameters
 parameters.default <- function(x, ...) {
-  rlang::abort(
-    glue(
-      "`parameters` objects cannot be created from objects ",
-      "of class `{class(x)[1]}`."
-    )
+  cli::cli_abort(
+    "{.cls parameters} objects cannot be created from {.obj_type_friendly {x}}."
```
Comment (hfrick) [inline R/parameters.R:16]: Nice!

**Review by hfrick — COMMENTED** [review]
Woop! Bye bye `paste0`! 👋 

Would you please add tests for the messages that don't have one yet? I spotted an instance where the call has not yet been threaded through and another one with a test missing and I assume there might be more that are still missing a test.

File: R/grids.R (hunk 1 above)
Comment (EmilHvitfeldt) [reply R/grids.R:258]: I can thread through the call. but this code as far as I can seen isn't possible to reach naturally. It is only applied to the output of functions like `make_max_entropy_grid()` and `make_latin_hypercube_grid()` which returns

**Comment by EmilHvitfeldt** [comment]
I went in add added the missing tests where i could find them

**Review by hfrick — APPROVED** [review]
Thank you! Having more tests is super helpful. 

dials has quite a lot of code paths so I'm not surprised you found another piece of probably dead code. I'll make an issue for it, thank you for threading the call through for now!

**Pull request merged**